### PR TITLE
feat: add endpoint to update metadata properties

### PR DESCRIPTION
**Pull request by jfcalvo**
# Description

This PR adds the following changes:

* Add a new endpoint `PATCH /api/v1/metadata-properties/{metadata_property_id}` allowing the update of metadata properties. Right now only `title` and `visible_for_annotators` attributes are allowed in the update.
* `title` has been added to metadata properties.
* `description` has been removed from metadata properties.

**Type of change**

- [x] New feature (non-breaking change which adds functionality)

**How Has This Been Tested**

- [x] Adding unit tests for the new endpoint.
- [x] Running unit tests using PostgreSQL.

**Checklist**

- [ ] I added relevant documentation
- [x] follows the style guidelines of this project
- [x] I did a self-review of my code
- [ ] I made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [ ] I filled out [the contributor form](https://tally.so/r/n9XrxK) (see text above)
- [x] I have added relevant notes to the CHANGELOG.md file (See https://keepachangelog.com/)

### CODE REVIEW

File: src/argilla/server/contexts/datasets.py [hunk 1]
```diff
@@ -264,6 +265,19 @@ async def create_metadata_property(
     return metadata_property
 
 
+async def update_metadata_property(
+    db: "AsyncSession",
+    metadata_property: MetadataProperty,
+    metadata_property_update: MetadataPropertyUpdate,
+):
+    # TODO: Do we need some additional step to reindex on the search engine?
```
Comment (jfcalvo) [inline src/argilla/server/contexts/datasets.py]: @frascuchon the question in the TODO is for you. 😄

File: src/argilla/server/alembic/versions/7cbcccf8b57a_create_metadata_properties_table.py [hunk 2]
```diff
@@ -30,31 +30,21 @@
 
 
 def upgrade() -> None:
-    # ### commands auto generated by Alembic - please adjust! ###
```
Comment (jfcalvo) [inline src/argilla/server/alembic/versions/7cbcccf8b57a_create_metadata_properties_table.py]: @gabrielmbmb I will update the PR so the migration is automatically made by Alembic.

Comment (jfcalvo) [reply src/argilla/server/alembic/versions/7cbcccf8b57a_create_metadata_properties_table.py]: Done.

I have changed the migration so we use alembic automatic migration and added some missing stuff on the models. 

You can see the changes on this commit: https://github.com/argilla-io/argilla/pull/3952/commits/639f07ceef42bef5dca1ee23771bc148ca17cee9

File: src/argilla/server/contexts/datasets.py (hunk 1 above)
Comment (frascuchon) [reply src/argilla/server/contexts/datasets.py]: No search engine reindex required since the "updatable" attributes does not affect to the search index

**Review by gabrielmbmb — APPROVED** [review]

**Pull request merged**